### PR TITLE
fix: useless menu components are now directives

### LINKED ISSUE #597: menu markup looks odd
#### Is this a bug, enhancement, or feature request?
enhancement

#### Briefly describe your proposal.
The current `menu` markup is:
```
<fd-menu>
    <fd-menu-list>
        <li fd-menu-item>
            <a [attr.href]="'#'">Option 1</a>
        </li>
        <li fd-menu-item>
            <a [routerLink]="'#'">Option 2</a>
        </li>
        <li fd-menu-item>
            Option 3
        </li>
    </fd-menu-list>
</fd-menu>
```
I think there is a room for simplifying/flattening it and instead of component `fd-menu-list` we can have an attribute directive on a `<ul>` element. Especially since we do the same for `<li>` with the directive `fd-menu-list`

#### Which versions of Angular and Fundamental NGX are affected? (If this is a feature request, use current version.)
angular 7.3
fundamental-ngx 0.2.0

### PR DESCRIPTION
**Pull request by MattL75**
#### Please provide a link to the associated issue.
fixes #597 

#### Please provide a brief summary of this pull request.
fd-menu-list, fd-menu-title and fd-menu-item have been turned to directives. It allows more control over the elements while conserving functionality and style.

@droshev This is a breaking change and should be documented somewhere. Maybe on the wiki?

### CODE REVIEW

**Comment by droshev** [comment]
@MattL75 can we avoid having the wrapping element `fd-menu` and rely only on directives? It will flatten the markup.

File: library/src/lib/menu/menu-list.directive.spec.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { MenuListDirective } from './menu-list.directive';
+
+describe('MenuListDirective', () => {
+    it('should create an instance', () => {
```
Comment (droshev) [inline library/src/lib/menu/menu-list.directive.spec.ts]: is there a way we can add more test than just instantiating the directive?

**Comment by MattL75** [comment]
> @MattL75 can we avoid having the wrapping element `fd-menu` and rely only on directives? It will flatten the markup.

We could, but personally I think it would be confusing to use. Having the top-level element as a directive seems much clearer.

File: library/src/lib/menu/menu-list.directive.spec.ts (hunk 1 above)
Comment (MattL75) [reply library/src/lib/menu/menu-list.directive.spec.ts]: all it does is add a class to the host

Comment (droshev) [reply library/src/lib/menu/menu-list.directive.spec.ts]: can we check that? otherwise the test looks like just targeting simple code coverage.

**Comment by droshev** [comment]
🚢 it looks good.

**Pull request merged**